cargo-cyclonedx/src/cli.rs

```diff
@@ -154,21 +158,19 @@ impl Args {
             Target::SingleTarget(target_string)
         });
 
-        let output_options = match (cdx_extension, prefix) {
-            (Some(cdx_extension), Some(prefix)) => Some(OutputOptions {
-                cdx_extension,
-                prefix,
-            }),
-            (Some(cdx_extension), _) => Some(OutputOptions {
-                cdx_extension,
-                prefix: Prefix::default(),
-            }),
-            (_, Some(prefix)) => Some(OutputOptions {
-                cdx_extension: CdxExtension::default(),
-                prefix,
-            }),
-            (_, _) => None,
-        };
+        let output_options =
+            if cdx_extension.is_none() && prefix.is_none() && !self.no_target_suffix {
+                None
+            } else {
+                Some(OutputOptions {
+                    cdx_extension: cdx_extension.unwrap_or_default(),
```

Comment:
This comment doesn't concern this line in particular but in https://cyclonedx.org/specification/overview/#recognized-file-patterns it is outlined how a valid CycloneDX file is called and this PR violates that.

If it's not `bom.xml` anymore, it should be `bom_x86_64-unknown-linux-gnu.cdx.xml`